tsl/src/bgw_policy/reorder_api.c

```diff
@@ -183,9 +181,11 @@ policy_reorder_add(PG_FUNCTION_ARGS)
 													  existing->fd.config)),
 												  NameGetDatum(index_name))))
 		{
-			elog(WARNING,
-				 "could not add reorder policy due to existing policy on hypertable with different "
-				 "arguments");
+			ereport(WARNING,
+					(errmsg("reorder policy already exists for hypertable \"%s\"",
+							get_rel_name(ht_oid)),
+					 errdetail("A policy already exists with different arguments."),
+					 errhint("Remove the existing policy before adding a new one.")));
 			PG_RETURN_INT32(-1);
```

Comment:
AIUI, this is an error from the user perspective (we're not "proceeding with the operation") so maybe this should be an error? I would argue that the add policy function should not have an implicit "if not exists" last, which this is equivalent to.

Reply:
I agree, but had this discussion previously with @gayyappan and there was some disagreement. I don't remember the arguments, but it might have something to do with always giving back a return value. Let's leave behavioral changes for another PR unless there's a bug.

Reply:
Ok. If it is a contention point we should not take it as part of this PR.